src/uicomponents/PokemonLookup.tsx

```diff
@@ -0,0 +1,1097 @@
+    }])
+}
+
+
+
+type SearchOption = {
+    name: string,
+    type: "Pokémon" | "Moves" | "Ability" | "Type" | "Custom",
+}
+
+const allOptions: SearchOption[] = [];
+for (let specie of genSpecies) {
+    allOptions.push({name: specie.includes("Flab") ? "Flabebe" : specie, type: "Pokémon"});
+}
+for (let type of genTypes) {
+    allOptions.push({name: type, type: "Type"});
+}
+for (let ability of genAbilities) {
+    allOptions.push({name: ability, type: "Ability"});
+}
+for (let move of genMoves) {
+    allOptions.push({name: move, type: "Moves"});
+}
+
+function checkSpeciesForFilters(species: PokemonData, filters: SearchOption[], translationKey: any) {
+    for (let filter of filters) {
+        switch (filter.type) {
+            case "Moves":
+                const moveFilter = filter.name as MoveName;
+                let moveFilterResult = false;
+                for (let move of species.moves) {
+                    if (moveFilter === move.name) {
+                        moveFilterResult = true;
+                        break;
+                    }
+                }
+                if (!moveFilterResult) {
+                    return false;
+                }
+            break;
+            case "Ability":
+                const abilityFilter = filter.name as AbilityName;
+                let abilityFilterResult = false;
+                for (let ability of species.abilities) {
+                    if (abilityFilter === ability.name) {
+                        abilityFilterResult = true;
+                        break;
+                    }
+                }
+                if (!abilityFilterResult) {
+                    return false;
+                }
+            break;
+            case "Type":
+                const typeFilter = filter.name as TypeName;
+                let typeFilterResult = false;
+                for (let type of species.types) {
+                    const uppercaseType = type[0].toUpperCase() + type.slice(1) as TypeName; // need to fix capitalization in assets branch
+                    if (typeFilter === uppercaseType) {
+                        typeFilterResult = true;
+                        break;
+                    }
+                }
+                if (!typeFilterResult) {
+                    return false;
+                }
+            break;
+            case "Custom":
+                const operatorPrecedence = {
+                    "&&": 1,
+                    "||": 0
+                };
+                const andOperators = ["and", "&&", "&", ","];
+                const orOperators = ["or", "||", "|"];
+                const validOperators = andOperators.concat(orOperators);
+
+                const filterComponents = 
+                    filter.name.toLowerCase().split(/(?<=\sand\s|\sor\s|&&|\|\||&|\||,|\(|\))|(?=\sand\s|\sor\s|&&|\|\||&|\||,|\(|\))/i).map((s) => s.trim())
```

Comment:
I was able to get the page to load on Safari by removing this line, so iOS is probably not happy with the regex.

Reply:
Oops, I didn't realize that Safari doesn't support lookbehind regex